stdlib.sh

```diff
@@ -1701,6 +2340,45 @@ function __STDLIB_API_1_std::getfilesection() { # {{{
 
 ###############################################################################
 #
+# stdlib.sh - Helper functions - Get key/value pairs off of Windows-style .ini
+#                                sections.
+#
+###############################################################################
+
+function __STDLIB_API_1_std::getKeyValuePairs() { # {{{
+	local param_section="${1:-}"
+	local retList=""
+
+	retList="$( echo "${param_section:-1}" | sed -r 's/#.*$// ; /^[^[:space:]]+\.[^[:space:]]+\s*=/s/\./_/' | grep -Ev '^\s*$' | sed -r 's/\s*=\s*/=/' )"
+
+	respond "${retList}"
+
+	return 0
+} # __STDLIB_API_1_std::getKeyValuePairs # }}}
+
+
+###############################################################################
+#
+# stdlib.sh - Helper functions - Get key/value pairs off of Windows-style .ini
+#                                sections.
+#
+###############################################################################
+
+function __STDLIB_API_1_std::getValueByKey() { # {{{
+	local param_pairList="${1:-}"
+	local param_key="${2:-}"
+	local retValue=""
+
+	retValue="$( grep -E "^${param_key:-}=" <<<"${param_pairList:-1}" | cut -d'=' -f2- )"
```

Comment:
The idea is that you would be passing the output from the previous function and therefore an already filtered pair, but defensive coding is probably better, yes.